File: app/src/main/java/info/nightscout/androidaps/plugins/general/nsclient/data/NSMbg.java
package info.nightscout.androidaps.plugins.general.nsclient.data;

import org.json.JSONException;
import org.json.JSONObject;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import info.nightscout.androidaps.logging.L;

public class NSMbg {
    private static Logger log = LoggerFactory.getLogger(L.NSCLIENT);
    public long date;
    public double mbg;
    public String json;

    public NSMbg(JSONObject json) {
        try {
            date = json.getLong("mills");
            mbg = json.getDouble("mgdl");
            this.json = json.toString();
        } catch (JSONException e) {
            log.error("Unhandled exception", e);
            log.error("Data: " + json.toString());
        }
    }
}

File: app/src/main/java/info/nightscout/androidaps/plugins/sync/nsclient/data/NSMbg.kt
package info.nightscout.androidaps.plugins.sync.nsclient.data

import info.nightscout.androidaps.utils.JsonHelper
import org.json.JSONObject

class NSMbg(val json: JSONObject) {

    var date: Long = 0
    var mbg: Double = 0.0

    init {
        date = JsonHelper.safeGetLong(json, "mills")
        mbg = JsonHelper.safeGetDouble(json, "mgdl")
    }

    fun id(): String? = JsonHelper.safeGetStringAllowNull(json, "_id", null)
    fun isValid(): Boolean = date != 0L && mbg != 0.0
}